src/libcore/iter/traits/collect.rs

```diff
@@ -341,11 +341,24 @@ pub trait Extend<A> {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     fn extend<T: IntoIterator<Item = A>>(&mut self, iter: T);
+
+    /// Extends a collection with exactly one element.
+    #[unstable(feature = "extend_one", issue = "none")]
+    fn extend_one(&mut self, item: A) {
+        self.extend(Some(item));
+    }
+
+    /// Reserves capacity in a collection for the given number of additional elements.
+    ///
+    /// The default implementation does nothing.
+    #[unstable(feature = "extend_one", issue = "none")]
+    fn extend_reserve(&mut self, _additional: usize) {}
```

Comment:
The argument name appears in rustdoc so it would be better to name it in a way that can be a canonical name for that argument across all downstream impls of Extend. `_additional` does highlight that it's probably unused by the default impl but I think that is less important.

```suggestion
    fn extend_reserve(&mut self, additional: usize) {
        let _ = additional;
    }
```